### DISCUSSION TITLE
Integrated Terrain System for Godot

### DISCUSSION THREAD

**Discussion by bongblender** [Redot-Engine/redot-proposals#63] [Ideas]
Feature Request: Godot could greatly benefit from an integrated terrain system by drawing inspiration from existing open plugins like HTerrain and MTerrain. Starting with core functionalities—such as height mapping, multi-layered texturing, and LOD management—based on these plugins would provide an effective foundation. By gradually building on top of this base, Godot could develop a robust, efficient terrain system natively, reducing dependency on external tools and enhancing 3D environment creation for developers. IT WOULD DEFINETLY GIVE BOOST TO REDOT, NO DOUBT.

**Comment by Arctis-Fireblight** [comment]
We will not be including a terrain system in Redot, however, we have built one into our other project ReX Engine. We hope to have a public release of that soon!
